pkg/things/thing.go

```diff
@@ -36,6 +39,17 @@ var (
 	ErrUnauthorised = errors.New("unauthorised")
 )
 
+type commandFormat uint8
+
+const (
+	commandFormatJOSE commandFormat = iota
```

Comment:
Can we use `contentType` directly rather than map between command format and content type? So add a constant for `content-type`, something like:
```
type ContentType string

const (
	applicationJSON ContentType = "application/json"
	applicationJOSE ContentType = "application/jose"
)
...
    AccessToken(tokenID string, type ContentType, payload io.Reader) (reply []byte, err error)
    Attributes(tokenID string, type ContentType, payload io.Reader, names []string) (reply []byte, err error)
```

Reply:
I did consider this when I implemented it but the old C dev in me was thinking about the greater efficiency of comparing integers rather than comparing strings. I like simplicity of this and we are going to have a whole efficiency push at a later date.